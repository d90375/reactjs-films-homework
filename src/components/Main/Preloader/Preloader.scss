.preloader {
  display: flex;
  flex-direction: column;
  align-items: center;
}

.preloader__picture {
  .preloader__child {
    width: (36px / 2);
    height: (36px / 2);
    background-color: #0aaee4;
    border-radius: 100%;
    display: inline-block;
    animation: preloader__picture 1.4s ease-in-out 0s infinite both;
  }

  .preloader__bounce-1 {
    margin-right: 9px;
    animation-delay: 0.32s;
  }

  .preloader__bounce-2 {
    margin-right: 9px;
    animation-delay: (0.32s / 2);
  }
}

@keyframes preloader__picture {
  0%,
  80%,
  100% {
    transform: scale(0);
  }

  40% {
    transform: scale(1);
  }
}

.preloader__text {
  opacity: 0.35;
  color: #242b2d;
  font-family: "Helvetica Neue", serif;
  font-size: 22px;
  font-weight: 500;
  line-height: 33px;
  text-transform: uppercase;
  letter-spacing: 0.88px;
  margin-bottom: 108px;
}
