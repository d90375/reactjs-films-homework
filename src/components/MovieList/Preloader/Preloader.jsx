import React from "react";

import styles from "./preloader.scss";

const Preloader = () => {
  return (
    <div className={styles.wrap}>
      <div className={styles.preloader}>
        <div className={styles.picture}>
          <div className={`${styles.child} ${styles.bounce1}`} />
          <div className={`${styles.child} ${styles.bounce2}`} />
          <div className={styles.child} />
        </div>
        <div className={styles.text}>LOADING</div>
      </div>
    </div>
  );
};

export default Preloader;
