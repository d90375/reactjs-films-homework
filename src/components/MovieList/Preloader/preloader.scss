.wrap {
  align-items: center;
  display: flex;
  flex-direction: column;
  height: 80vh;
  justify-content: center;
  width: 100%;
}

.preloader {
  align-items: center;
  display: flex;
  flex-direction: column;
}

.picture {
  .child {
    animation: picture 0.5s ease-in-out 0s infinite both;
    background-color: #0aaee4;
    border-radius: 100%;
    display: inline-block;
    height: (28px / 2);
    width: (28px / 2);
  }

  .bounce1 {
    animation-delay: 0.32s;
    margin-right: 9px;
  }

  .bounce2 {
    animation-delay: (0.32s / 2);
    margin-right: 9px;
  }
}

@keyframes picture {
  0%,
  80%,
  100% {
    transform: scale(0);
  }

  40% {
    transform: scale(1);
  }
}

.text {
  color: #242b2d;
  font-family: "HelveticaNeue", serif;
  font-size: 1rem;
  font-weight: 500;
  letter-spacing: 0.88px;
  line-height: 33px;
  margin-bottom: 54px;
  opacity: 0.35;
  text-transform: uppercase;
}
