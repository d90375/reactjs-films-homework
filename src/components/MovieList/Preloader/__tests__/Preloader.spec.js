import React from "react";
import { shallow } from "enzyme";
import Preloader from "../Preloader";

describe("Preloader component", () => {
  let component;

  it("should render Preloader component", () => {
    component = shallow(<Preloader />);
    expect(component).toMatchSnapshot();
  });
});
