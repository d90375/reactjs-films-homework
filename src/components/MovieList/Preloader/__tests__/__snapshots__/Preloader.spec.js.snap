// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Preloader component should render Preloader component 1`] = `
<div
  className="wrap"
>
  <div
    className="preloader"
  >
    <div
      className="picture"
    >
      <div
        className="child bounce1"
      />
      <div
        className="child bounce2"
      />
      <div
        className="child"
      />
    </div>
    <div
      className="text"
    >
      LOADING
    </div>
  </div>
</div>
`;
